test/Feature/InsertSpec.hs

```diff
@@ -145,7 +145,7 @@ spec = do
           `shouldRespondWith` ResponseMatcher {
             matchBody    = Just inserted
           , matchStatus  = 201
-          , matchHeaders = ["Location" <:> [str|/json?data=eq.{"foo":"bar"}|]]
+          , matchHeaders = ["Location" <:> "/json?data=eq.%7B%22foo%22%3A%22bar%22%7D"]
```

Comment:
No, JSON keys aren't working for me, whether it's URL-encoded or not, and whether it's a list or an object. I'll add tests for those.